platform-services/src/std_platform_services.rs

```diff
@@ -15,14 +17,48 @@
 use super::result;
 
 use getrandom;
+use nix::{
+    time,
+    sys::time::TimeValLike,
+};
 
 /// Fills a buffer, `buffer`, with random bytes sampled from the random number
 /// source provided by the host operating system, as provided by `getrandom`.
 ///
 /// This is for use with "freestanding-execution-engine".
-pub fn platform_getrandom(buffer: &mut [u8]) -> result::Result {
+pub fn platform_getrandom(buffer: &mut [u8]) -> result::Result<()> {
     if let Ok(_) = getrandom::getrandom(buffer) {
-        return result::Result::Success;
+        return result::Result::Success(());
     }
     result::Result::UnknownError
 }
+
+/// Returns the clock resolution in nanoseconds.
+///
+/// This is for use with "freestanding-execution-engine".
```

Comment:
I think this comment should be omitted: it may find use elsewhere and then the comment will be out of date.

Reply:
Note that the same comment appears in at least one other place.

Reply:
Agreed, should we remove it from `getrandom()` too?

Reply:
Yeah, I think so.